### PR TITLE
Spinner html

### PR DESCRIPTION
**Pull request by arschmitz**
This pr moves the `_uiSpinnerHtml` and `_buttonHtml` extension points into backcompat

### CODE REVIEW

File: tests/unit/spinner/deprecated.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+define( [
+	"jquery",
+	"ui/spinner"
+], function( $, testHelper ) {
```
Comment (scottgonzalez) [inline tests/unit/spinner/deprecated.js]: Drop `testHelper`, it's not used and it's wrong.

File: tests/unit/spinner/deprecated.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+} );
+
+test( "markup structure - custom", function( assert ) {
+	expect( 2 );
+	var element = $( "#spin" ).spinner(),
+		spinner = element.spinner( "widget" ),
+		up = spinner.find( ".ui-spinner-up" );
+
+	assert.hasClasses( spinner, "ui-spinner ui-widget ui-widget-content spin-wrap", "_uiSpinnerHtml overides default markup" );
+	assert.hasClasses( up, "ui-spinner-button ui-spinner-up ui-widget spin-up", "_uiButtonHtml overides default markup" );
```
Comment (scottgonzalez) [inline tests/unit/spinner/deprecated.js]: This is minor, but the method name should be followed by parens in the messages.

File: ui/spinner.js [hunk 3]
```diff
@@ -214,13 +214,25 @@ return $.widget( "ui.spinner", {
...
-				.append( this._buttonHtml() );
+
+				// Add buttons
+				.append(
+				"<a>" +
+					"<span>&#9650;</span>" +
+				"</a>" +
+				"<a>" +
+					"<span>&#9660;</span>" +
+				"</a>" );
```
Comment (scottgonzalez) [inline ui/spinner.js]: Indent the markup, move the closing paren to its own line.

File: ui/spinner.js [hunk 4]
```diff
@@ -533,4 +531,38 @@ return $.widget( "ui.spinner", {
 	}
 } );
 
+// DEPRECATED
+// TODO: switch return back to widget declaration at top of file when this is removed
+if ( $.uiBackCompat !== false ) {
+
+	// Backcompat for dialogClass option
```
Comment (scottgonzalez) [inline ui/spinner.js]: Gotta love the dialog classes on the spinner ;-)

File: ui/spinner.js [hunk 5]
```diff
@@ -533,4 +531,38 @@ return $.widget( "ui.spinner", {
...
+	$.widget( "ui.spinner", $.ui.spinner, {
+		_enhance: function(){
+			this.uiSpinner = this.element
```
Comment (scottgonzalez) [inline ui/spinner.js]: Why is assignment to `this.uiSpinner` happening here?

Comment (scottgonzalez) [inline ui/spinner.js]: Oh, I see, we're not leveraging the existing method. Ignore my question.

**Comment by arschmitz** [comment]
Ready for another review

**Comment by jzaefferer** [comment]
This looks good to me.

**Pull request merged**